ddtrace/internal/datadog/profiling/stack_v2/src/sampler.cpp

```diff
@@ -56,17 +58,63 @@ Sampler::get()
     return instance;
 }
 
+void
+_stack_v2_atfork_child()
+{
+    // The only thing we need to do at fork is to propagate the PID to echion
+    // so we don't even reveal this function to the user
+    _set_pid(getpid());
+}
+
+__attribute__((constructor)) void
+_stack_v2_init()
+{
+    _stack_v2_atfork_child();
+}
+
 void
 Sampler::one_time_setup()
 {
     _set_cpu(true);
     init_frame_cache(echion_frame_cache_size);
-    _set_pid(getpid());
+
+    // It is unlikely, but possible, that the caller has forked since application startup, but before starting echion.
+    // Run the atfork handler to ensure that we're tracking the correct process
+    _stack_v2_atfork_child();
+    pthread_atfork(nullptr, nullptr, _stack_v2_atfork_child);
 
     // Register our rendering callbacks with echion's Renderer singleton
     Renderer::get().set_renderer(renderer_ptr);
 }
 
+void
+Sampler::register_thread(uintptr_t id, uint64_t native_id, const char* name)
+{
+    const std::lock_guard<std::mutex> thread_info_guard{ thread_info_map_lock };
```

Comment:
Could you add a comment saying that `thread_info_map_lock` and `thread_info_map` are defined in 'echion/threads.h'

Reply:
Sure thing

Reply:
Fixed in c83e1e35f